### PR TITLE
Fixing tBTCv2 system tests

### PR DESCRIPTION
**Pull request by dimpar**
This PR updates/fixes tBTCv2 integration and system tests. 

- Fix prerequisites by adding the correct test setup to run the tests. This includes some changes in the `system-test.yml` and the deployment scripts
- Introducing an actual Random Beacon contract with available groups to perform work for the integration test
- Run the full circle of deposit and redemption (no TBTC Vault / no minting)

### CODE REVIEW

**Comment by michalinacienciala** [comment]
I've started the System Tests job on the feature branch from this PR: https://github.com/keep-network/tbtc-v2/actions/runs/3677255433.

**Comment by michalinacienciala** [comment]
> I've started the System Tests job on the feature branch from this PR: https://github.com/keep-network/tbtc-v2/actions/runs/3677255433.

@dimpar , unfortunately the System Tests workflow that I run on the `fixing-system-tests` branch has been automatically canceled (due to 6h job execution timeout). It looks that the job hanged on the `Setup scenario prerequisites` step. Could you take a look and see if you can find an explanation of this behavior?

**Comment by nkuba** [comment]
@michalinacienciala could you please try again? I pushed a fix in 3f9e361b48a582935e878418b5b1eef7ec9ce590

**Comment by michalinacienciala** [comment]
> @michalinacienciala could you please try again? I pushed a fix in https://github.com/keep-network/tbtc-v2/commit/3f9e361b48a582935e878418b5b1eef7ec9ce590

Started: https://github.com/keep-network/tbtc-v2/actions/runs/3749883573 🤞

**Comment by michalinacienciala** [comment]
> Started: https://github.com/keep-network/tbtc-v2/actions/runs/3749883573 🤞

@nkuba , still failing 😞 (on different stage that before, though)

**Comment by michalinacienciala** [comment]
Next tests execution after the fixes: https://github.com/keep-network/tbtc-v2/actions/runs/3756700180

**Comment by michalinacienciala** [comment]
> Next tests execution after the fixes: https://github.com/keep-network/tbtc-v2/actions/runs/3756700180

@nkuba, still failing...

**Comment by pdyraga** [comment]
Heads up, the changes from #442 may implicate some updates to system tests. I did not apply changes in #442 given the system tests are failing and I wouldn't be able to test if the changes were correct.

**Comment by nkuba** [comment]
> Heads up, the changes from #442 may implicate some updates to system tests. I did not apply changes in #442 given the system tests are failing and I wouldn't be able to test if the changes were correct.

The updates were introduced in https://github.com/keep-network/tbtc-v2/pull/449

I invoked the system tests with the latest changes: https://github.com/keep-network/tbtc-v2/actions/runs/3824709740

**Comment by nkuba** [comment]
Another try: https://github.com/keep-network/tbtc-v2/actions/runs/3828141616/jobs/6513401598

File: solidity/hardhat.config.ts [hunk 1]
```diff
@@ -99,6 +99,10 @@ const config: HardhatUserConfig = {
       // deploy stub contracts in tests.
       allowUnlimitedContractSize: process.env.TEST_USE_STUBS_TBTC === "true",
     },
+    localhost: {
```
Comment (pdyraga) [inline solidity/hardhat.config.ts:102]: Let's name this network `system_tests` to be explicit it is about system tests and not, for example, local developer's box. Note that this is the approach from #471.

Comment (dimpar) [reply solidity/hardhat.config.ts]: Already changed by merging `main`.

File: system-tests/.gitignore [hunk 2]
```diff
@@ -1,2 +1,7 @@
+# Hardhat
+/build/
 /cache/
+/export/
 /external/
+/typechain/
+/export.json
```
Comment (tomaszslabon) [inline system-tests/.gitignore:7]: There is a convention to have an empty line at the end of all files.

File: solidity/test/integration/data/bls.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const blsData = {
+  // data generated using master secret key 123
+  secretKey: 123,
+
+  // altbn128 public key for secret key 123. For implementation details refer to bls_test.go
+  groupPubKey:
+    "0x1f1954b33144db2b5c90da089e8bde287ec7089d5d6433f3b6becaefdb678b1b2a9de38d14bef2cf9afc3c698a4211fa7ada7b4f036a2dfef0dc122b423259d01659dc18b57722ecf6a4beb4d04dfe780a660c4c3bb2b165ab8486114c464c621bf37ecdba226629c20908c7f475c5b3a7628ce26d696436eab0b0148034dfcd",
```
Comment (tomaszslabon) [inline solidity/test/integration/data/bls.ts:7]: Could we split the hex-string into multiple lines for easier reading?
```suggestion
     "0x1f1954b33144db2b5c90da089e8bde287ec7089d5d6433f3b6becaefdb678b1b2a9de3" +
    "8d14bef2cf9afc3c698a4211fa7ada7b4f036a2dfef0dc122b423259d01659dc18b57722" +
    "ecf6a4beb4d04dfe780a660c4c3bb2b165ab8486114c464c621bf37ecdba226629c20908" +
    "c7f475c5b3a7628ce26d696436eab0b0148034dfcd",
    ```

File: system-tests/.gitignore (hunk 2 above)
Comment (dimpar) [reply system-tests/.gitignore]: Sure [f2e884f](https://github.com/keep-network/tbtc-v2/pull/421/commits/f2e884f2d18612de52065cf053b4282550cc9ac8)

File: .github/workflows/system-tests.yml [hunk 4]
```diff
@@ -17,17 +17,24 @@ jobs:
...
         env:
           WALLET_PUBLIC_KEY: ${{ secrets.SYSTEM_TESTS_WALLET_PUBLIC_KEY }}
-          ETH_NETWORK: localhost
+          ETH_NETWORK: system_tests
         run: |
           yarn install
-          npx hardhat node --export export.json &
+          yarn clean
+          npx hardhat node --no-deploy &
+          USE_EXTERNAL_DEPLOY=true npx hardhat deploy --network $ETH_NETWORK --export export.json
```
Comment (tomaszslabon) [inline .github/workflows/system-tests.yml:37]: Shouldn't we update [the readme on system tests](https://github.com/keep-network/tbtc-v2/blob/main/system-tests/README.adoc).

Comment (dimpar) [reply .github/workflows/system-tests.yml:37]: Good point [0cfa14f](https://github.com/keep-network/tbtc-v2/pull/421/commits/0cfa14f9276420ecb5248cab17d4320d2ce76c5b)

**Review by tomaszslabon — APPROVED** [review]

**Pull request merged**